django/forms/boundfield.py

```diff
@@ -133,23 +133,9 @@ def value(self):
         Returns the value for this BoundField, using the initial value if
         the form is not bound or the data otherwise.
         """
-        if not self.form.is_bound:
-            data = self.form.initial.get(self.name, self.field.initial)
-            if callable(data):
-                if self._initial_value is not UNSET:
-                    data = self._initial_value
-                else:
-                    data = data()
-                    # If this is an auto-generated default date, nix the
-                    # microseconds for standardized handling. See #22502.
-                    if (isinstance(data, (datetime.datetime, datetime.time)) and
-                            not self.field.widget.supports_microseconds):
-                        data = data.replace(microsecond=0)
-                    self._initial_value = data
-        else:
-            data = self.field.bound_data(
-                self.data, self.form.initial.get(self.name, self.field.initial)
-            )
+        data = self.initial
+        if self.form.is_bound:
+            data = self.field.bound_data(self.data, data)
```

Comment:
Makes sense.

I have reverted to using the old branching style in the PR. I can investigate why the `else` is different as a separate issue/pr. Thanks.